crates/sui-storage/src/write_ahead_log.rs

```diff
@@ -134,6 +139,9 @@ where
 // A WriteAheadLog implementation built on rocksdb.
 pub struct DBWriteAheadLog<C> {
     log: DBMap<TransactionDigest, C>,
+    // We use two tables, because if we instead have one table mapping digest -> (C, u32), we have
+    // to clone C to make a tuple ref to pass to insert.
+    retry_count: DBMap<TransactionDigest, u32>,
```

Comment:
What could lead to a retry (same question as before). Is this expected or exceptional?

Reply:
A retry count for a tx is incremented every time `pop_one_recoverable_tx()` returns that tx. So this is caused by any transient error or crash that leaves an uncommitted tx in the WAL.

Reply:
I think it is 100% worth it - otherwise, if someone submits a tx that reliable crashes the validator, we will be stuck in a crash loop until someone manually removes the bad tx from the db.